src/planner/plan/append_extend.cpp

```diff
@@ -122,6 +128,71 @@ void Planner::appendNonRecursiveExtend(const std::shared_ptr<NodeExpression>& bo
     }
 }
 
+void Planner::appendRecursiveExtendAsGDS(const std::shared_ptr<NodeExpression>& boundNode,
+    const std::shared_ptr<NodeExpression>& nbrNode, const std::shared_ptr<RelExpression>& rel,
+    ExtendDirection direction, LogicalPlan& plan) {
+    // GDS pipeline
+    auto recursiveInfo = rel->getRecursiveInfo();
+    auto graphEntry =
+        graph::GraphEntry(recursiveInfo->node->getTableIDs(), recursiveInfo->rel->getTableIDs());
+    auto functionSet = VarLenJoinsFunction::getFunctionSet();
+    KU_ASSERT(functionSet.size() == 1);
+    auto gdsFunction = functionSet[0]->constPtrCast<GDSFunction>()->copy();
+    auto bindData = std::make_unique<RJBindData>(boundNode, nbrNode, recursiveInfo->lowerBound,
```

Comment:
I wonder if you should bind the function with a dummy direction during binder, and set the direction when we're actually planning for it, though I'm not sure if planner is feasible to take in bound match pattern mixed with gds call.

Reply:
This could be an option. Though I'm not very convinced of the benefit of this approach. So I'll leave this as it is for now.